include/RAJA/policy/cuda/teams.hpp

```diff
@@ -98,7 +95,63 @@ struct LaunchExecute<RAJA::expt::cuda_launch_t<async, 0>> {
       RAJA_FT_END;
     }
 
-    // return resources::EventProxy<resources::Cuda>(cuda_res);
+  }
+
+  template <typename BODY_IN>
+  static resources::EventProxy<resources::Resource>
+  exec(RAJA::resources::Resource res, LaunchContext const &ctx, BODY_IN &&body_in)
```

Comment:
Should this take a Cuda resource instead of a type erased resource?

Reply:
I would be in favor of doing it with the next iteration of the API.